### PR TITLE
fix: display appropriate value when tx sent to self

### LINKED ISSUE #1856: When you send to yourself then transaction detail amount is only fee
When you send to yourself then transaction detail amount is only fee.
I don't know if this is feature or bug but i sent 1$ and not 3 cents, that's fee.

image

### PR DESCRIPTION
**Pull request by robertbrada**
fix: https://github.com/trezor/trezor-suite/issues/1856

I changed the **Amount** value we show to the user when he sends a transaction to himself. In this PR we show _"the actual value user sent + fee"_, not just the fee (even though the user lost from his wallet portfolio just the fee). 

This value could be equal to the `totalInput` value (right?), but I encountered a problem that `totalInput` is not defined for _Ethereum_ network. Is that a bug or a valid behavior? If it's a bug, it should be fixed and this issue could be solved in a cleaner way as well. 

For the above-mentioned reasons, I solved this from the other side: `amountSentToSelf = totalOutput + tx.fee`. 
You can see the numbers I computed on the provided screenshots, which capture "**Send ETH to myself**" transaction.

image

image

However, because `tx.fee` is a string, I had to convert it to a float and then to an integer first, in order to prevent issues with summing up float number (is this an ok approach?).

_Note:_
I didn't try to solve the situation when the user sends some funds to himself and some funds to somebody else. If somebody thinks that the way how we show this information should be changed, let me know.

### CODE REVIEW

File: packages/suite/src/components/suite/modals/TransactionDetail/components/AmountDetails/index.tsx
Comment (keraf) [inline packages/suite/src/components/suite/modals/TransactionDetail/components/AmountDetails/index.tsx]: I don't think a ternary is necessary here as the variable already has an empty string assigned to it when initialized. I would transform this into an if and set its value if the condition is true.

**Review by keraf — COMMENTED** [review]
Just a minor comment, the rest looks good to me.

**Review by keraf — APPROVED** [review]
Looks good to me now but waiting for @szymonlesisz 's review before merging as he know that part better than me :smile:

**Comment by robertbrada** [comment]
I found out that the suite (or backend?) does not recognize "self" transactions when Ethereum tokens are transfered, even though it should (if that gets fixed, this PR will introduce new error). 

I will adjust the code for the situation in which the user sends token to himself. In such scenario the displayed amount will be just the `tokenAmount`, not `totalOutput + fee`.

**Comment by robertbrada** [comment]
> I found out that the suite (or backend?) does not recognize "self" transactions when Ethereum tokens are transfered, even though it should (if that gets fixed, this PR will introduce new error).
> 
> I will adjust the code for the situation in which the user sends token to himself. In such scenario the displayed amount will be just the `tokenAmount`, not `totalOutput + fee`.

I found out that the `type: "self"` attribute is assigned to the token transaction, not the transaction itself, which technically makes sense. That's the reason why Suite doesn't handle "_Sent TOKEN to myself_" when checking for `tx.type==="self"`. So I am not if we should handle this somehow in the UI or leave it as it is.

image

**Comment by szymonlesisz** [comment]
i would rather do this change in lower level than this component. what about transaction list? we don't want to use fixed amounts there as well?
here?
https://github.com/trezor/trezor-suite/blob/develop/packages/suite/src/utils/wallet/transactionUtils.ts#L336-L369
or even lower, in blockchain-link where the transaction is parsed.
https://github.com/trezor/trezor-suite/blob/develop/packages/blockchain-link/src/workers/blockbook/utils.ts#L131
im going to do some changes there soon, so i might also implement this calculation

in ETH & RIPPLE there are no "inputs/outputs" this is only suite ui unification. those networks are not "utxo based" so technically there is no such a thing as "inputs".

token transactions may be ambiguous. technically this is the only ETH transaction that may have multiple outputs, depending on smart contract that you are woking with. However 90%+ of transactions in the suite will be basic transfers from A to B.
Basically by sending token you are not sending to the recipient address (B) but to the token address (T) 
Token tx contains additional data for the smart contract on address T. letting him know that you are transferring from address A to address B
this is why transaction type is assigned to the the token transfer, not to the whole transaction.

**Comment by slowbackspace** [comment]
since I closed original issue, I am also closing this PR 🙇

**Pull request closed without merging**
